staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/meta.go

```diff
@@ -115,8 +115,20 @@ func (meta *ListMeta) GetSelfLink() string               { return meta.SelfLink
 func (meta *ListMeta) SetSelfLink(selfLink string)       { meta.SelfLink = selfLink }
 func (meta *ListMeta) GetContinue() string               { return meta.Continue }
 func (meta *ListMeta) SetContinue(c string)              { meta.Continue = c }
-func (meta *ListMeta) GetRemainingItemCount() int64      { return meta.RemainingItemCount }
-func (meta *ListMeta) SetRemainingItemCount(c int64)     { meta.RemainingItemCount = c }
+
+func (meta *ListMeta) GetRemainingItemCount() int64 {
+	if meta.RemainingItemCount != nil {
+		return *meta.RemainingItemCount
+	}
+	return 0
+}
+
+func (meta *ListMeta) SetRemainingItemCount(c int64) {
+	if meta.RemainingItemCount == nil {
+		meta.RemainingItemCount = new(int64)
```

Comment:
this implementation means that all calls to SetRemainingItemCount will result in a serialized output. If we need to distinguish between set and unset, we should have this:
```
GetRemainingItemCount() *int64
SetRemainingItemCount(c *int64)
```

and calling `SetRemainingItemCount(nil)` should remove the field from unstructured implementations